tools/appveyor.psm1

```diff
@@ -35,8 +35,15 @@ function Invoke-AppVeyorInstall {
     $dotNetCoreSDKVersion = $globalDotJson.sdk.version
     # don't try to run this script on linux - we have to do the negative check because IsLinux will be defined in core, but not windows
     if (-not ((dotnet --version).StartsWith($dotNetCoreSDKVersion)) -and ! $IsLinux ) {
-        [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12 # https://github.com/dotnet/announcements/issues/77
-        Invoke-WebRequest 'https://dot.net/v1/dotnet-install.ps1' -OutFile dotnet-install.ps1
+        try {
+            $originalSecurityProtocol = [Net.ServicePointManager]::SecurityProtocol
```

Comment:
You could add an if condition that only changes if `[Net.SecurityProtocolType]::Tls12` is not currently added.

Reply:
Isn't this what the bitwise operator `-bor` already does?

Reply:
my point is you could detect that no change is needed... not set `$originalSecurityProtocol` and not restore it if it is not set.